packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -75,6 +77,13 @@ function findRoutesAndCreateOptimalPath(
     fees: FeeSchedule,
     neonRouterNumSamples: number,
 ): Path | undefined {
+    // Currently the rust router is unable to handle 1 base unit sized quotes and will error out
+    // To avoid flooding the logs with these errors we just return an insufficient liquidity error
+    // which is how the JS router handles these quotes today
+    if (input.eq(ONE_BASE_UNIT)) {
```

Comment:
But it can handle, like, 2?

Reply:
yes, strangely enough 😂  

It does a sanity assertion that for every sample index i, `samples[i] <= samples[i + 1]`, which should work as long as the sampled amounts are equal or increasing. But when all samples are the same value (eg `1`) this assertion fails. I expect it's due to some f64 precision shenanigans but I haven't had time to investigate 🤷‍♂️